### DISCUSSION TITLE
Replacing /nsm with larger disk -- LVM mirror or copy/replace?

### DISCUSSION THREAD

**Discussion by abesinger** [Security-Onion-Solutions/securityonion#6945] [Unsupported Versions]
I want to replace /nsm with a larger disk.  I've already read [The SO section on adding disk](https://docs.securityonion.net/en/2.3/new-disk.html).

So there's a couple ways of doing it, what are people's thought's on one version the other? I've got the new disk attached to the system.  I could either:

A) stop services, move /nsm, copy to new disk, mount new disk as /nsm

or

B) Use LVM to add the disk to the existing volume group and use [LVM Mirroring](https://www.casesup.com/category/knowledgebase/howtos/how-to-migrate-lvm-to-new-storage) to copy, and then remove the original disk and extend the existing XFS partition to use all of the new disk.

A is a little more simple, but requires the system to be basically down while migrating data. 

B is a cleaner approach and allows the system to remain running, but could maybe affect performance?

Thoughts?

**Comment by petiepooo** [comment]
If /nsm is part of your root filesystem, I would suggest method A.  Trying to switch your boot disk is possible, but fraught with missed details and grub rescue prompts.  Trust me; I found success, but learned more than I cared to about grub rescue mode, initramfs prompts, and Dell's perccli utility.

On the other hand, if /nsm is already a separate logical volume on its own disk, method B should work great for you.  Running pvmove should not cause huge performance issues, but i would avoid excessive queries and target a quiet period anyway.  Read the man pages carefully and double-check your typing before you hit enter.

if your nsm and root LVs share the same PV, I would suggest keeping root on the original PV and moving nsm to your new one.  See note above about the perils of switching boot disks.

**Comment by abesinger** [comment]
Thanks, that's useful information.

`/nsm` is not part of the root filesystem, but it's not on it's own disk, it's a separate logical volume in the same volume group with `/`, `/tmp`, and `/swap`. (It's how SO configured the disk)

Which actually tells me that I need to pursue and entirely different path.  I wanted to move data and recover the original  disk, but that would involve also moving `/` and `/boot` and dealing with grub.

**Reply by petiepooo** [reply]
That's what I was concerned about.  It's possible, but complicated.  If grub rescue scares you (and it should), I'd avoid it.

However, if you move /nsm onto its own disk, there's nothing stopping you from expanding root into the newly freed space, or creating another volume of your choice.  I typically make /var/log a separate volume, as that's usually why root fills up, at least in my experience...  You might also consider making a separate volume just for Elasticsearch data if it has the space and IO bandwidth.

**Comment by abesinger** [comment]
grub's not really what I'm scared of.  It's running on vmware (by someone else), I don't know how vmware deals with boot blocks, and I don't want to spend the time to figure it out.  So I had them resize the disk, and I've just got to resize through LVM and then XFS.  Going to make a snapshot first just to be sure.
